package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs

```diff
@@ -266,67 +268,84 @@ public class GeometrySubsets {
+
+      if (options.meshOptions.points == ImportMode.Import && usdMesh.points != null)
+      {
+        if (changeHandedness)
+        {
+          newVertices = new Vector3[usdMesh.points.Length];
+          for (int i = 0; i < usdMesh.points.Length; i++)
+          {
+            newVertices[i] = UnityTypeConverter.ChangeBasis(usdMesh.points[i]);
           }
         }
+        
+        if (IsThereFaceVaryingAttributes(usdMesh, options))
+        {
+          newVertices = SanitizePointsForUnity(newVertices ?? usdMesh.points, usdMesh.faceVertexIndices);
+          newIndices = Enumerable.Range(0, usdMesh.faceVertexIndices.Length).ToArray();
+          arePointsSanitized = newVertices.Length == usdMesh.faceVertexIndices.Length;
+        }
 
         if (usdMesh.faceVertexIndices != null) {
           // Annoyingly, there is a circular dependency between vertices and triangles, which makes
           // it impossible to have a fixed update order in this function. As a result, we must clear
           // the triangles before setting the points, to break that dependency.
           unityMesh.SetTriangles(new int[0] { }, 0);
         }
-        unityMesh.vertices = usdMesh.points;
-      }
-
-      //
-      // Purpose.
-      //
-
-      // Deactivate non-geometry prims (e.g. guides, render, etc).
-      if (usdMesh.purpose != Purpose.Default) {
-        go.SetActive(false);
+        unityMesh.vertices = newVertices ?? usdMesh.points;
       }
 
       //
       // Mesh Topology.
       //
 
-      // TODO: indices should not be accessed if topology is not requested, however it may be
-      // needed for facevarying primvars; that special case should throw a warning, rather than
-      // reading the value.
-      int[] originalIndices = new int[usdMesh.faceVertexIndices == null
-                                      ? 0
-                                      : usdMesh.faceVertexIndices.Length];
-      // Optimization: only do this when there are face varying primvars.
-      if (usdMesh.faceVertexIndices != null) {
-        Array.Copy(usdMesh.faceVertexIndices, originalIndices, originalIndices.Length);
-      }
-
-      if (options.meshOptions.topology == ImportMode.Import && usdMesh.faceVertexIndices != null) {
+      if (options.meshOptions.topology == ImportMode.Import && usdMesh.faceVertexIndices != null)
+      {
         Profiler.BeginSample("Triangulate Mesh");
-        if (options.meshOptions.triangulateMesh) {
-          // Triangulate n-gons.
-          // For best performance, triangulate off-line and skip conversion.
-          if (usdMesh.faceVertexIndices == null) {
+        if (options.meshOptions.triangulateMesh)
+        {
+          // TODO: I think that vertex indices and counts are mandatory attributes for a prim with a Mesh schema,
+          //       we could have a generic validation phase for that (generic = uniform checks of mandatory
+          //       attributes for any schema). 
```

Comment:
Broken assets often are missing data that should be required :/

Reply:
Yes, but what I mean is that maybe... we could have a function that given a prim and it's schema, validate that all mandatory attributes are valid, and if not, ignore the prim... or something like that... and avoid too much "if blabla != null" in the code.